### DISCUSSION TITLE
Platformer Behaviour: Drop/Let go of ladder

### DISCUSSION THREAD

**Discussion by AtomicCrocStudios** [4ian/GDevelop#2368] [Archive]
# See https://github.com/4ian/GDevelop/discussions/2339 instead.

## Description
Real *simple* request, a new platformer control/option that allows the player to 'Drop or Let Go' of a ladder. 
This would allow us to de-attach the player whilst on the ladder for various reasons, in my case when they're at the bottom and on solid ground.

## Solution suggested
**Describe the solution**
When activated, the player would let go of the ladder and set isOnLadder() to false. If the player is in the air when activated, they would fall to the ground (much like when you jump excluding the going up part).

I tried doing this is JS, but it looks like nothing is exposed to let me do this.

**Comment by 4ian** [comment]
Indeed, much like there is a "release key" for when the object is grabbing a platform, there should be the same thing (an action) for releasing a ladder.
This will need a update to the behavior (platformerobjectruntimebehavior.ts).

**Comment by krunkster** [comment]
Unsolicited workaround: you should be able to deactivate the platform behavior of the ladder object temporarily to achieve this effect.

**Comment by AtomicCrocStudios** [comment]
> Unsolicited workaround: you should be able to deactivate the platform behavior of the ladder object temporarily to achieve this effect.

Thanks for the suggestion, I do have a work around working via JS. Just thought it might be something useful to have within the actions much like releasing from a platform :)

**Comment by Silver-Streak** [comment]
@4ian This appears to be a duplicate of the request (since converted to a discussion) here: https://github.com/4ian/GDevelop/discussions/2339

Additionally, it's on the Trello board here: https://trello.com/b/qf0lM7k8/gdevelop-roadmap?menu=filter&filter=label:Feat%3A%20platformer%20engine

No clue if these can be merged, but one of these discussions should be closed. 

(I do, however, agree it's a good idea)

**Reply by 4ian** [reply]
Indeed, locked and the discussion can continue on https://github.com/4ian/GDevelop/discussions/2339
